homeassistant/components/squeezebox/coordinator.py

```diff
@@ -68,5 +75,34 @@ def _prepare_status_data(self, data: dict) -> dict:
             else None
         )
 
+        # Updates
+        # newversion str not always present
+        # Sample text:-
+        # 'A new version of Logitech Media Server is available (8.5.2 - 0). <a href="updateinfo.html?installerFile=/var/lib/squeezeboxserver/cache/updates/logitechmediaserver_8.5.2_amd64.deb" target="update">Click here for further information</a>.'
+        # '<ul><li>Version %s - %s is available for installation.</li><li>:og in to your computer running Logitech Media Server (%s).</li><li>Execute <code>%s</code> and follow the instructions.</li></ul>'
+        data[STATUS_SENSOR_NEWVERSION] = (
+            self.newversion_regex_pre.sub(
+                "",
+                self.newversion_regex_post.sub(
+                    "",
+                    self.newversion_regex_windows_pre.sub(
+                        "",
+                        self.newversion_regex_windows_post.sub(
+                            "", data[STATUS_SENSOR_NEWVERSION]
+                        ),
+                    ),
+                ),
+            )
+            if STATUS_SENSOR_NEWVERSION in data
+            else data[STATUS_QUERY_VERSION]
+        )
```

Comment:
So I checked with the core team, and this falls under [ADR-4](https://github.com/home-assistant/architecture/blob/master/adr/0004-webscraping.md), which means it's parsing HTML and thus is not allowed in core. Originally I was doubting if this was considered parsing HTML, hence I double checked with the rest.

Is it possible to get squeezebox to adopt a JSON response for this?

Reply:
Ok so this is deliberately constructed (we only ever delete) such that if any pattern patching fails this would all still work, it just looks prettier for the end user. So I could remove the regex's here and the code would work as intended. The contents of the string is irrelevant just the fact there is one.

Reply:
Could you contextualize the variables?

Reply:
This is a JSON response with embedded html in the returned string...

Reply:
Sure these are all returned(on not) as part of JSON serverstatus (returns a dict of values) call so in this case if STATUS_SENSOR_NEWVERSION is present this means there is a new version of LMS.  This then contains a localized string about the update which in some installs returns with some embedded html. data[STATUS_QUERY_VERSION] is well defined and just contains the current version.  That what you where after?

Reply:
What is `data[STATUS_QUERY_VERSION]` then?

Reply:
data[STATUS_QUERY_VERSION] is well defined and just contains the current version again its part of the status call data. eg "8.5.2"

Reply:
Is worth me mentioning the substitution to delete, is just to present a nicer string to the end user, that why I was doing it we can go with raw or sub the hole string for a known value (though we loose the to be version with that).

Reply:
Oh you mean you would replace the actual new version with just a string? We should set that to the newest version, but without HTML parsing

Reply:
Option 1
Parse string this is what I have now.

Option 2
```
data[STATUS_SENSOR_NEWVERSION] = "New Version" if STATUS_SENSOR_NEWVERSION in data
            else data[STATUS_QUERY_VERSION]
```
 I cant get at the version no itself unless I passed the string return from LMS, but I Can indicate there is a new version

Option 3
```
if STATUS_SENSOR_NEWVERSION not in data:
            data[STATUS_QUERY_VERSION]
```
Raw

Reply:
If you're not allowed to do 1) then I think 2) is likely enough most of the time - I doubt most people look at the version number but if they're interested, look at the links for details of what's changed.  I guess the only time I notice the version is when a version change on a release track changes - e.g. when "stable" soon changes from 8.5.x to 9.0.0.

Doesn't newversion returned from the json always contain a string at the front with the version number - before you get to the html?

I see 

```
A new version of Logitech Media Server is available (8.5.3 - 1726474675).
```
as the start of that string.  Would is still be considered parsing html if you just threw away everything after the ```).``` before doing anything else - we don't even know or care that there sometimes happens to be html on some platforms then?

Reply:
I was thinking about this on the cycle home by placing it in the release summary buy just extracting the first sentence.  that way its not used in any logic and we can stick new version in the updated version slot very simple low risk and reduces it to one regex. It doesn't matter that some versions(windows) will have some html in either at that point as its rendered, @joostlek shall i put that up for people to look at?

Reply:
Ok update entities now show a simple state based on the presence of updates and additional information has been moved into the release summary.

Reply:
So how to we move this forward, the only thing the code currently does now is extract the first sentence for the release summary text for the server updates, plugin update is now used raw in the release summary, update logic for both only relies on the presence of response in the status JSON api call.

Reply:
All seems to work fine as far as I can see and avoids parsing the html.

Reply:
@rajlaud you happy with this one?

Reply:
Yes, this makes sense to me.